### PR TITLE
refactor(ui): main flow mobile design review fix part 1

### PR DESCRIPTION
**Pull request by simeng-li**
## Summary

This PR includes following updates 

1. Update the sign-in page divider i18n content phrase key. Only apply 'continue with' when social is a primary sign-in method, apply 'or' for all the rest.

2. Remove username and password format validation in sign-in form, should only validate format in create account flow. 

3. Replace the clear icon for input field. 
image

4. Remove Error border for passcode input fields
image

5. Do not show error highlight for confirm password mismatch error
image

6. Show clear icon for PasswordInput and ConfirmPasswordInput field in the Create Account Form. Using general input component to replace the old password input component. 
image

7. Update password-less sign-in confirm modal confirm text
image

8. Divider space update

9. Update the sign-in page create account link behavior
 - link should follow the users primary sign-in method settings
 - do not show the link if social is set as primary sign-in method

## Linear Issue Reference

log-2340

## Testing

test locally 
@logto-io/eng

### CODE REVIEW

File: packages/ui/src/containers/SocialSignIn/PrimarySocialSignIn.tsx [hunk 1]
```diff
@@ -19,15 +19,15 @@ const PrimarySocialSignIn = ({ className, isPopup = false, onSocialSignInCallbac
   const [showAll, setShowAll] = useState(false);
   const { invokeSocialSignIn, socialConnectors } = useSocial({ onSocialSignInCallback });
   const isOverSize = socialConnectors.length > defaultSize;
-  const displayAll = showAll || isPopup || !isOverSize;
+  const displayAll = isPopup || !isOverSize;
 
   const displayConnectors = useMemo(() => {
-    if (displayAll) {
+    if (displayAll || showAll) {
```
Comment (demonzoo) [inline packages/ui/src/containers/SocialSignIn/PrimarySocialSignIn.tsx:25]: nit: `displayAll` and `showAll` are a little bit confusing. I don't know, maybe we should rename the former to `shouldDisplayAll`?

**Review by demonzoo — APPROVED** [review]
LGTM

File: packages/ui/src/components/Icons/ClearIcon.tsx [hunk 2]
```diff
@@ -1,10 +1,10 @@
 import React, { SVGProps } from 'react';
 
-import CloseIcon from '@/assets/icons/close-icon.svg';
+import IconClear from '@/assets/icons/clear-icon.svg';
 
 const ClearIcon = (props: SVGProps<SVGSVGElement>) => (
-  <svg width="24" height="24" viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg" {...props}>
-    <use href={`${CloseIcon}#close-icon`} />
+  <svg width="18" height="18" viewBox="0 0 18 18" xmlns="http://www.w3.org/2000/svg" {...props}>
+    <use href={`${IconClear}#clear`} />
```
Comment (demonzoo) [inline packages/ui/src/components/Icons/ClearIcon.tsx:7]: What does the `#clear` do?

Comment (simeng-li) [reply packages/ui/src/components/Icons/ClearIcon.tsx:7]: Some how use href required a id hash anchor to identify the SVG, This whole approach is some how fuzzy. Need to refine some day.

File: packages/ui/src/containers/SocialSignIn/PrimarySocialSignIn.tsx (hunk 1 above)
Comment (simeng-li) [reply packages/ui/src/containers/SocialSignIn/PrimarySocialSignIn.tsx]: make sense.  updated

**Pull request merged**